src/components/icon/icon.ts

```diff
@@ -27,9 +28,18 @@ const iconCache = new Map<string, Promise<SVGResult>>();
 export default class SlIcon extends ShoelaceElement {
   static styles: CSSResultGroup = styles;
 
+  private initialRender = false;
+
   /** Given a URL, this function returns the resulting SVG element or an appropriate error symbol. */
-  private static async resolveIcon(url: string): Promise<SVGResult> {
+  private async resolveIcon(url: string, library?: IconLibrary): Promise<SVGResult> {
```

Comment:
I had to move this method to the instance as otherwise, making the load and error observable was way too complicated.

Reply:
That should be fine. I dont think anyone is out here calling `SlIcon.resolveIcon()` anyways.